### PR TITLE
Spinners galore (AKA fix all the things with spinners)

### PR DESCRIPTION
**Pull request by acburdine**
Note: this PR is based on #24 - once that one is merged I'll rebase this one but I thought i'd get the changes in so that they can be tested.

Tested this with denali as a child addon, going to do some more tests with other existing addons. Denali itself won't work out of the box just by adding `isDevelopingAddon=true`.... needs a separate PR to make it work as a development addon (PR: https://github.com/denali-js/denali/pull/340).

### CODE REVIEW

**Comment by acburdine** [comment]
👍 now that #24 is merged I'll rebase this

File: .travis.yml [hunk 1]
```diff
@@ -4,6 +4,9 @@ node_js:
   - "6.9"
   - "7"
   - "8"
+before_script:
```
Comment (acburdine) [inline .travis.yml]: :trollface:

File: lib/project.ts [hunk 2]
```diff
@@ -234,9 +235,7 @@ export default class Project {
         lint: this.lint,
         audit: this.audit
       });
-      // TODO: revisit to fix promise issues
-      // tslint:disable-next-line:no-floating-promises
-      addonProject.watch({ onBuild: options.onBuild, outputDir: addonDist });
+      return addonProject.watch({ onBuild: options.onBuild, outputDir: addonDist });
```
Comment (davewasmer) [inline lib/project.ts]: Feels like I'm missing something here - seems like `project.watch()` doesn't actually return anything, and the outer `Bluebird.each` here isn't actually awaited. So why the return and each here?

File: lib/project.ts [hunk 3]
```diff
@@ -245,19 +244,20 @@ export default class Project {
...
       this.finishBuild(results, options.outputDir);
       await spinnerStart;
-      await spinner.succeed(`${ this.pkg.name } build complete (${ timer.stop() }s)`);
+      await spinner.succeed(`${ this.pkg.name } build complete (${ timer.stop() }s)`, this.pkg.name);
       spinnerStart = null;
       options.onBuild(this);
     });
     watcher.on('error', async (error: any) => {
-      await spinner.fail('Build failed');
+      await spinnerStart;
```
Comment (davewasmer) [inline lib/project.ts]: Ah, was this the core problem?

File: lib/project.ts (hunk 2 above)
Comment (acburdine) [inline lib/project.ts]: It does return a promise, and it makes it so the watchers are started serially (i.e. we don't have two watchers of child addons starting at the same time). As to whether making the watchers start serially actually _does_ anything I'm not sure.

**Pull request merged**